File: src/main/java/io/mfedirko/common/validation/ValidReCaptcha.java
package io.mfedirko.common.validation;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.*;

@Documented
@Constraint(validatedBy = ReCaptchaValidator.class)
@Target( { ElementType.FIELD })
@Retention(RetentionPolicy.RUNTIME)
public @interface ValidReCaptcha {
    String message() default "Please complete the reCaptcha";
    Class<?>[] groups() default {};
    Class<? extends Payload>[] payload() default {};
}

File: src/main/kotlin/io/mfedirko/common/validation/ValidReCaptcha.kt
package io.mfedirko.common.validation

import jakarta.validation.Constraint
import kotlin.reflect.KClass

@MustBeDocumented
@Constraint(validatedBy = [ReCaptchaValidator::class])
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class ValidReCaptcha(
    val message: String = "Please complete the reCaptcha",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Any>> = []
)